File: unit-3/sprint-2/day-2/we/Download a File/app/src/main/java/com/example/downloadafile/MainActivity.java
package com.example.downloadafile;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button mBtnDownloadFile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mBtnDownloadFile = findViewById(R.id.btnSave);
        Intent intent = new Intent(this, DownloadService.class);
        mBtnDownloadFile.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(intent);
            }
        });
    }
}

File: unit-4/sprint-2/day-1/assignments/Content providers assignment/app/src/main/java/com/example/contentprovidersassignment/MainActivity.kt
package com.example.contentprovidersassignment

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.SimpleCursorAdapter
import com.example.contentprovidersassignment.adaptor.showAdaptor
import com.example.contentprovidersassignment.model.MessageModel
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.row.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var adaptor: SimpleCursorAdapter
    private var Sms : MutableList<MessageModel> = ArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnInbox.setOnClickListener(this)
        btnSentBox.setOnClickListener(this)
    }

    override fun onClick(item: View?) {
        if(item == btnInbox){
            val inbox = Uri.parse("content://sms/inbox")
            val requiredColumn = arrayOf("_id", "address", "body")
            val cursor = contentResolver
            val c = cursor.query(inbox, requiredColumn, null, null, null)

            adaptor = SimpleCursorAdapter(
                this, R.layout.row, c, arrayOf("body", "address"), intArrayOf(
                    R.id.Name, R.id.Number)
            )
            listview.adapter = adaptor
        }

        if(item == btnSentBox){
            val inbox = Uri.parse("content://sms/sent")
            val requiredColumn = arrayOf("_id", "address", "body")
            val cursor = contentResolver
            val c = cursor.query(inbox, requiredColumn, null, null, null)

            adaptor = SimpleCursorAdapter(
                this, R.layout.row, c, arrayOf("body", "address"), intArrayOf(
                    R.id.Name, R.id.Number)
            )
            listview.adapter = adaptor
        }

    }
}